### PR TITLE
docs: update `DataFrame` docstrings to include the errors section

### PR DESCRIPTION
**Pull request by arwas11**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary):
  - `to_gbq`: screen/6fahEb9CJ7r69yc
  - `to_parquet`: screen/3kBwqd5AHRTpP5W
  - `insert`: screen/QXBv9pWR8M9Toxo
  - `drop`: screen/9Si5uK9NhFjudc2
  - `set_index`: screen/aYEyVMgfeNH9nyk
  - `reorder_levels`: screen/4xRALnZrpUTMrSN
  - `swaplevel`: screen/7j9Zu4JJyzfnXMp
  - `droplevel`: screen/C5myDpnkGqLCCYe
  - `dropna`: screen/49hfZbyTjZqixVe
  - `isin`: screen/6fTGLaFYBe2NHGM
  - `sort_values`: screen/6FKiveqddSpZ8dz
  - `sort_index`: screen/3oHDsq5EHngLJgB
  - `combine`: screen/3vJBqsWf7TKe7dt
  - `explode`:  screen/3oHDsq5EHngLJgB
  - `update`: screen/5dgpp9iqZpoLTXE
  - `groupby`: screen/8tSm9QYnQvLiixU
  - `map`: screen/BmfrwCH79QDeDsX
  - `join`: screen/FqZvzHrnGN9KvqX
  - `merge`: screen/57k9UzYy2MZ6oMr
  - `apply`: screen/4S3csoNQ9cHnxF5
  - `nlargest`: screen/BxvVoyrVm524yCu
  -  `nsmallest`: screen/6hu4XG9ZknqNdcQ
  - `cumsum`: screen/7WT2AwmpCYyn2E9
  - `cumprod`: screen/6pjvBi3PY9wXS7e
  - `describe`: screen/5SU9DPxx338oUwh
  - `dot`: screen/Bt88eX4xUeQNLvC
  - `sample`: screen/9fYVqia2grZpeye
  - `filter`: screen/5oGofrU87PayQJu
 

Fixes #296390934 🦕

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 1]
```diff
@@ -3813,8 +3885,17 @@ def explode(
...
+                Exploded lists to rows of the subset columns;
                 index will be duplicated for these rows.
+
+        Raises:
+            ValueError:
+                * If columns of the frame are not unique.
+                * If specified columns to explode is empty list.
+                * If specified columns to explode have not matching count of elements rowwise in the frame.
+            KeyError:
+                If incorecct column names are provided
```
Comment (tswast) [inline third_party/bigframes_vendored/pandas/core/frame.py:3898]: Typo

```suggestion
                If incorrect column names are provided
```

**Review by tswast — APPROVED** [review]
Thank you!

**Comment by shobsi** [comment]
Are there screenshots of the documentation (the "Appropriate docs were updated" section in the description) after this change? E.g. https://github.com/googleapis/python-bigquery-dataframes/pull/1081

**Comment by arwas11** [comment]
> Are there screenshots of the documentation (the "Appropriate docs were updated" section in the description) after this change? E.g. #1081

@shobsi  Thanks for pointing that out! I added the screenshots.

**Pull request merged**